src/concepts/rendering.md

```diff
@@ -77,3 +77,231 @@ This 4 minute talk about `IMGUI` is also tangentially relevant.
 - **Architecture design considerations**: With `ratatui`, out of the box, there's little to no help
   in organizing large applications. Ultimately, the decision on structure and discipline rests with
   the developer to be principled.
+
+## How does Ratatui work?
+
+The smallest unit of data in Ratatui is a [`Cell`].
+
+```rust
+pub struct Cell {
```

Comment:
I don't know that I like showing the internals like this as it means one extra place to update if we make any changes. Whereas showing the method calls and linking to the docs seems more useful.

Reply:
What if I did this? I just want to show that a `Cell` struct in essence is a `String` + the `Style`. I don't need the code to be accurate here.

```rust
// pseudo code
struct Cell {
  symbol: String,
  style: Style,
}
```